src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,159 @@
+    private static final String PRIVATE_KEY_PEM_BOUNDARY = "PRIVATE KEY";
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final ClientDevicesAuthService clientDevicesAuthService;
+    private final CertificateManager certificateManager;
+    private final ExecutorService executorService;
+    private KeyPair keyPair;
+
+    /**
+     * Constructor.
+     *
+     * @param context                  operation continuation handler
+     * @param clientDevicesAuthService client device auth service
+     * @param executorService          executor service
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         ClientDevicesAuthService clientDevicesAuthService,
+                                                         ExecutorService executorService) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.clientDevicesAuthService = clientDevicesAuthService;
+        this.certificateManager = clientDevicesAuthService.getCertificateManager();
+        this.executorService = executorService;
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        //TODO: Unsubscribe to cert updates.
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public SubscribeToCertificateUpdatesResponse handleRequest(
+            SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+        return translateExceptions(() -> {
+            try {
+                clientDevicesAuthService.doAuthorizationForSubscribingToCertUpdates(serviceName);
+            } catch (AuthorizationException e) {
+                throw new UnauthorizedError(e.getMessage());
+            }
+            try {
+                String certificateType = getCertificateTypeFromCertificateOptions(
+                        subscribeToCertificateUpdatesRequest.getCertificateOptions());
+                if (CERT_TYPE_SERVER.equals(certificateType)) {
+                    final String csr = getCSRForServer();
+                    Consumer<X509Certificate> serverCertificateCallback = this::forwardServerCertificates;
+                    executorService.submit(() -> {
+                        try {
+                            TimeUnit.SECONDS.sleep(2);
+                            certificateManager.subscribeToServerCertificateUpdates(csr, serverCertificateCallback);
+                        } catch (KeyStoreException | CsrProcessingException | InterruptedException e) {
+                            logger.atError().log("Unable to subscribe to server certificate updates");
+                            throw new ServiceError(
+                                    String.format("Subscribe to server certificate update failed with error %s", e));
+                        }
+                    });
```

Comment:
why does this need to be done in a different thread? Subscribing to updates shouldn't take any time.